tsdb/shard_test.go

```diff
@@ -1696,6 +1697,45 @@ func TestMeasurementFieldSet_InvalidFormat(t *testing.T) {
 	}
 }
 
+func TestMeasurementFieldSet_ConcurrentSave(t *testing.T) {
+	var iterations int64
+	dir, cleanup := MustTempDir()
+	defer cleanup()
+
+	if testing.Short() {
+		iterations = 50
+	} else {
+		iterations = 200
+	}
+
+	path := filepath.Join(dir, "fields.idx")
+	mf, err := tsdb.NewMeasurementFieldSet(path)
+	if err != nil {
+		t.Fatalf("NewMeasurementFieldSet error: %v", err)
+	}
+	var wg sync.WaitGroup
+
+	wg.Add(4)
+	go testFieldMaker(t, &wg, mf, "cpu", "cvalue", iterations)
+	go testFieldMaker(t, &wg, mf, "cpu", "cvalue", iterations)
+	go testFieldMaker(t, &wg, mf, "cpu", "cvalue", iterations)
+	go testFieldMaker(t, &wg, mf, "cpu", "cvalue", iterations)
+	wg.Wait()
```

Comment:
I'd like to see some more assertions here that the final file ended up with the correct data.